doc/migration/v0.1.0-migration-guide.md

```diff
@@ -0,0 +1,268 @@
+listOps := &client.ListOptions{Namespace: memcached.Namespace, LabelSelector: labelSelector}
+err := r.client.List(context.TODO(), listOps, podList)
+
+// Get
+dep := &appsv1.Deployment{APIVersion: "apps/v1", Kind: "Deployment", Name: name, Namespace: namespace}
+err := sdk.Get(dep)
+// v0.1.0
+dep := &appsv1.Deployment{}
+err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, dep)
+```
+
+Lastly copy and intialize any other fields that you may have had in your `Handler` struct into the `Reconcile<Kind>` struct:
+
+```Go
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileMemcached{client: mgr.GetClient(), scheme: mgr.GetScheme(), foo: "bar"}
+}
+
+// ReconcileMemcached reconciles a Memcached object
+type ReconcileMemcached struct {
+    client client.Client
+    scheme *runtime.Scheme
+    // Other fields
+    foo string
+}
+```
+
+### Copy changes from main.go
+
+The main function for a `v0.1.0` operator in `cmd/manager/main.go` sets up the [Manager][manager-go-doc] which registers the custom resources and starts all the controllers.
+
+There is no need to migrate the SDK functions `sdk.Watch()`,`sdk.Handle()`, and `sdk.Run()` from the old `main.go` since that logic is now defined in a controller.
+
+However if there are any operator specific flags or settings defined in the old main file copy those over.
+
+If you have any 3rd party resource types registered with the SDK's scheme, then register those with the Manager's scheme in the new project. See how to [register 3rd party resources][register-3rd-party-resources].
+
+### Copy user defined files
+
+If there are any user defined pkgs, scripts, and docs in the older project, copy these files into the new project. 
+
+### Copy changes to deployment manifests
+For any updates made to the following manifests in the old project, copy over the changes to their corresponding files in the new project:
+- `tmp/build/Dockerfile` to `build/Dockerfile`
+- RBAC rules updates from `deploy/rbac.yaml` to `deploy/role.yaml`
+- `deploy/cr.yaml` to `deploy/crds/<group>_<version>_<kind>_cr.yaml`
+- `deploy/crd.yaml` to `deploy/crds/<group>_<version>_<kind>_crd.yaml`
+
+### Copy user defined dependencies
+
+For any user defined dependencies added to the old project's Gopkg.toml, copy and append them to the new project's Gopkg.toml.
+Run `dep ensure` to update the vendor in the new project.
+
+### Confirmation
+
+At this point you should be able to build and run your operator to verify that it works. See the [user-guide][user-guide-build-run] on how to build and run your operator.
+
+[v0.1.0-changes-doc]: ./v0.1.0-changes.md
+[v0.0.7-memcached-operator]: https://github.com/operator-framework/operator-sdk-samples/tree/aa15bd278eec0959595e0a0a7282a26055d7f9d6/memcached-operator
+[v0.1.0-memcached-operator]: https://github.com/operator-framework/operator-sdk-samples/tree/master/memcached-operator
```

Comment:
This should probably be locked to a specific commit so this doc stays consistent as we update the SDK over time